polkadot/api/src/lib.rs

```diff
@@ -174,7 +177,7 @@ impl<B: Backend> PolkadotApi for Client<B, NativeExecutor<LocalDispatch>>
 	fn check_id(&self, id: BlockId) -> Result<CheckedId> {
 		// bail if the code is not the same as the natively linked.
 		if self.code_at(&id)? != LocalDispatch::native_equivalent() {
-			bail!(ErrorKind::UnknownRuntime);
+			warn!("This node is out of date. Block authoring may not work correctly.")
```

Comment:
dangerous proposition. I think authorities would prefer to get slashed for being temporarily offline than risk accidentally losing their whole bond.

also means that when we update the runtime, if >2/3 of the authorities have this behavior, they will continue growing the chain with the wrong rules.

Reply:
(I get why this change is in this PR -- the `genesis.wasm` is a nuisance to work around, but we should just stop checking this stuff in and have an environment variable for when we want to use a specific runtime wasm!)

Reply:
to be honest we're going to have to introduce semantic versioning here anyway. it's impractical to ensure that a node is updated, restarted and where necessary migrated at *exactly* the block boundary when the runtime changes, so there will need to be changeover periods. bailing here is unreasonable.

Reply:
right, but there are different approaches to that, and given that the API can change between runtimes, it's likely we might do something based on linking multiple versions of the `polkadot-api` crate. Network, consensus, and other client-level code might change between runtimes as well.

Reply:
i'd rather just have semantic versioning of the API and be done with it.

Reply:
can we agree to classify that as beyond the scope of this PR and introduce that later?

Reply:
sure